lib/parachain/types.go

```diff
@@ -0,0 +1,403 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// ValidatorIndex Index of the validator. Used as a lightweight replacement of the `ValidatorId` when appropriate
+type ValidatorIndex uint32
+
+// ValidatorID The public key of a validator.
+type ValidatorID [sr25519.PublicKeyLength]byte
+
+// BlockNumber The block number type.
+type BlockNumber uint32
```

Comment:
I think we can move this to `dot/types/block.go`

Reply:
Thanks @dimartiro !

I believe we're looking to contain the types to its respective package. 

@timwu20 What do you think?

Reply:
honestly, having a separate type for block number is n't really necessary.
At most places we have just used uint32 directly.

If we do however want to add a separate type for this then `dot/types` is not a bad place, BlockNumber is something that could get used across the project.

Reply:
If we don't have any receiver methods for a custom type, I would agree with Kishan, and probably just use the primitive.

Reply:
What is a receiver method?

Reply:
I believe otherwise. It provides clearer intent and enhances code readability.

Reply:
Resolving this cause I need these types in dispute-coordinator. Since this is being merged to a feature branch, happy to continue the conversation in the future.

Reply:
Well, it looks like I can't merge this. Is there anything I can do to get this PR merged so that I can continue my work on dispute-coordinator?

Reply:
> I believe otherwise. It provides clearer intent and enhances code readability. Could you tell me why you think having a receiver makes a difference?

That's the criteria I use when creating custom types.  Unless I want to add a method to the type, I'm ok with just using primitives.  I can go either way in this case, but if we were to use the parachain package and use types that use `parachain.BlockNumber`, the user would have to cast any existing primitive or custom types to `parachain.BlockNumber`, which could be avoided by just using the primitive.